lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -222,6 +231,17 @@ def cert_action(module, iam, name, cpath, new_name, new_path, state,
             module.exit_json(changed=changed, msg='Certificate with the name %s already absent' % name)
 
 
+def load_data(cert, key, cert_chain):
+    # if paths are provided rather than lookups read the files and return the contents
+    if os.path.isfile(cert):
+        cert = open(cert, 'r').read().rstrip()
+    if os.path.isfile(key):
+        key = open(key, 'r').read().rstrip()
+    if os.path.isfile(cert_chain):
+        cert_chain = open(module.params.get('cert_chain'), 'r').read()
```

Comment:
Something else to consider.    These statements should probably also check the value is not None, to avoid issues with isfile.  TypeError: coercing to Unicode: need string or buffer, NoneType found

```
if cert_chain is not None and os.path.isfile(cert_chain):
```